### PR TITLE
warehouse abusive restriction

### PR DESCRIPTION
**Pull request by homer8173**
you can have twice a room1 in two different buildings

### CODE REVIEW

File: htdocs/install/mysql/tables/llx_entrepot.key.sql [hunk 1]
```diff
@@ -17,4 +17,4 @@
 -- ============================================================================
 
 
-ALTER TABLE llx_entrepot ADD UNIQUE INDEX uk_entrepot_label (ref, entity);
+ALTER TABLE llx_entrepot ADD UNIQUE INDEX uk_entrepot_label (ref, entity,fk_parent);
```
Comment (hregis) [inline htdocs/install/mysql/tables/llx_entrepot.key.sql:20]: @homer8173 
remove and recreate Unique Index in migration script

Comment (homer8173) [reply htdocs/install/mysql/tables/llx_entrepot.key.sql:20]: Don't worry, it's fixed for me
but this fix should also improve Dolibarr

Comment (homer8173) [reply htdocs/install/mysql/tables/llx_entrepot.key.sql:20]: Sorry, you want me to report it in migration script also ?

Comment (hregis) [reply htdocs/install/mysql/tables/llx_entrepot.key.sql:20]: @homer8173 
if you change only in "/mysql/tables/", this fix is only for fresh install !
add this in "/mysql/migration/7.0.0-8.0.0.sql" for dolibarr update : 

ALTER TABLE llx_entrepot DROP INDEX uk_entrepot_label;
ALTER TABLE llx_entrepot ADD UNIQUE INDEX uk_entrepot_ref_fk_parent (ref, entity, fk_parent);

Comment (homer8173) [reply htdocs/install/mysql/tables/llx_entrepot.key.sql:20]: @hregis thanks for advices !

**Comment by eldy** [comment]
The "ref" is the visible key for user that make object unique.
It is recommanded to include a prefix/suffix in your ref if you need to use similar name in different hierarchy

**Comment by homer8173** [comment]
I understand your objective of standardisation but what about a room1 in a driveway1 in a floor1 in a building1 , must i had a prefix/suffix too if they already got a diff on the parent ?

furethemore, the getnomurl for this object offer a parent prefix system , it will be difficult to understand something later
maybe ref and label are 2 different things and should be separated here

**Comment by eldy** [comment]
This is why there is a field "short location" that is label. Here, you can have duplicate value
Ref is unique like any ref and label can be similar to another emplacement on a different level
For you exemple, ref can be R1-D1-F1-B1 and label "Room 1"

**Pull request closed without merging**